hoomd/mpcd/pytest/test_collide.py

```diff
@@ -130,3 +131,528 @@ def test_run(self, small_snap, simulation_factory, cls, init_args):
+            central_flag = new_snap.particles.typeid == new_snap.particles.types.index(
+                "A"
+            )
+            constit_flag = new_snap.particles.typeid == new_snap.particles.types.index(
+                "B"
+            )
+
+            new_velo_central = new_snap.particles.velocity[central_flag]
+            new_velo_constituent = new_snap.particles.velocity[constit_flag]
+            new_velo_free = new_snap.particles.velocity[free_flag]
+            new_velo_mpcd = new_snap.mpcd.velocity
+
+            # solve for expected central particle velocity based on linear momentum
+            initial_momentum_nonrigid = (
+                np.sum(velo_mpcd, axis=0) + np.array([0, 1, 0]) * participated_flag
+            )
+            initial_momentum_md = np.array([0, 2, 0]) * total_mass
+            initial_momentum = initial_momentum_md + initial_momentum_nonrigid
+
+            final_momentum_nonrigid = (
+                np.sum(new_velo_mpcd * new_snap.mpcd.mass, axis=0)
+                + new_velo_free * participated_flag
+            )
+            final_linmom_md = initial_momentum - final_momentum_nonrigid
+            final_velocity_md = final_linmom_md / total_mass
+            assert np.allclose(final_velocity_md, new_velo_central)
+
+            # solve for expected angular momentum change based on solvent
+            # multiply expected change by 2 to get quaternion
+            # the free particle interacts with the first constituent
+            change_momentum_nonrigid = (new_velo_mpcd - velo_mpcd) * new_snap.mpcd.mass
+            change_momentum_nonrigid[0] += (
+                np.squeeze(np.subtract(new_velo_free, [0, 1, 0])) * participated_flag
+            )
+            expected_change_angmom_md = np.zeros(4)
+            expected_change_angmom_md[1:] = (
+                np.sum(
+                    np.cross(def_rigid["positions"], -1 * change_momentum_nonrigid),
+                    axis=0,
+                )
+                * 2
+            )
+            change_angmom_md = new_snap.particles.angmom[central_flag] - np.array(
+                [0, 0, 0, 0]
+            )
+            assert np.allclose(expected_change_angmom_md, change_angmom_md)
+
+            # check the constituent velocities match the central particle
+            # since orientation is stuck at [1, 0, 0, 0], angular velocity
+            # is 0.5 * real part of angmom / inertia.
+            new_angmom = new_snap.particles.angmom[central_flag]
+            omega = [
+                0 if i == 0 else 0.5 * a / i
+                for i, a in zip(properties_rigid["inertia"], new_angmom[0, 1:])
+            ]
+            expected_tangential_velocity = np.cross(omega, def_rigid["positions"])
+
+            expected_velocity = np.add(expected_tangential_velocity, new_velo_central)
+            assert np.allclose(new_velo_constituent, expected_velocity)
+
+
+def test_rigid_thermostat_error(small_snap, simulation_factory):
```

Comment:
Is it possible to parametrize these error checks? It seems like there is a lot of duplicated code.